### ISSUE TITLE
Currently CustomRun.Status.CustomRunStatusFields.Results field do not support all ResultTypes

### ISSUE THREAD

**Issue by aaron-prindle** [tektoncd/pipeline#7751]
In the latest version of CustomRun v1beta1, the CustomRun.Status.CustomRunStatusFields.Results field do not support all ResultTypes.  This is the CustomRunResult struct:

https://github.com/tektoncd/pipeline/blob/main/pkg/apis/run/v1beta1/customrunstatus_types.go#L70-L76
```
// CustomRunResult used to describe the results of a task
type CustomRunResult struct {
    // Name the given name
    Name string `json:"name"`
    // Value the given value of the result
    Value string `json:"value"`
}
```

Currently I believe this is a bug as it makes it not possible to do fully interop with v1 Task/TaskRuns w/ a Custom Task (CustomRun) as there is not a way to properly map all ResultTypes.  

The fix here would be update the CustomRunResult to support String, Array, and Object Result types making the supported Results similar/identical to the Results available to V1 Task/TaskRuns

**Comment by chitrangpatel** [comment]
I agree, this feels unintentional. We should be able to add it easily I think.
